### PR TITLE
Support for unquoted hyphenated identifiers on bigquery

### PR DESCRIPTION
**Pull request by jmhain**
This adds support for unquoted hyphenated identifiers on bigquery, e.g. `select * from foo-bar f join baz-qux b on f.id = b.id`. These are only allowed in contexts where they are unambiguous, such as in a `FROM` or `JOIN` clause. (There may be other cases where they are allowed, I only tested these two.)

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -5867,10 +5888,10 @@ impl<'a> Parser<'a> {
 
     /// Parse a possibly qualified, possibly quoted identifier, e.g.
     /// `foo` or `myschema."table"
-    pub fn parse_object_name(&mut self) -> Result<ObjectName, ParserError> {
+    pub fn parse_object_name(&mut self, in_table_clause: bool) -> Result<ObjectName, ParserError> {
```
Comment (alamb) [inline src/parser/mod.rs:5891]: Can you please add documentation to this function explaining the new paramter and some context of what it is used for?

File: tests/sqlparser_bigquery.rs [hunk 2]
```diff
@@ -878,6 +878,27 @@ fn parse_table_identifiers() {
...
+    test_table_ident_err("foo-`bar`");
+    test_table_ident_err("`foo`-bar");
+    test_table_ident_err("foo-123a");
+    test_table_ident_err("foo - bar");
+    test_table_ident_err("123-bar");
+    test_table_ident_err("bar-");
+}
+
+#[test]
+fn parse_hyphenated_table_identifiers() {
```
Comment (alamb) [inline tests/sqlparser_bigquery.rs:897]: Can you also include some whole query level error tests as well

like that you can't do something like this (or at least document in tests that it parses to a subtraction?

```
select foo-bar.* from foo-bar
select foo-bar.x
```

**Review by alamb — COMMENTED** [review]
Thank you for the contribution @jmhain  -- this is wild

> This adds support for unquoted hyphenated identifiers on bigquery, e.g. select * from foo-bar f join baz-qux b on f.id = b.id. These are only allowed in contexts where they are unambiguous, such as in a FROM or JOIN clause. (There may be other cases where they are allowed, I only tested these two.)

Wow the craziness of sql dialects never ceases to amaze me. Big query, why? LOL

**Comment by jmhain** [comment]
@alamb Thanks for the prompt feedback! I pushed a couple commits incorporating your suggestions.

> Wow the craziness of sql dialects never ceases to amaze me. Big query, why? LOL

lol this was my reaction as well

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks again @jmhain

**Pull request merged**